### PR TITLE
ENH: Python 3.7 support

### LINKED ISSUE #531: Support Python 3.7
For CoLab

### PR DESCRIPTION
**Pull request by thewtex**
Closes #531

### CODE REVIEW

**Review by bnmajor — CHANGES_REQUESTED** [review]

File: itkwidgets/integrations/__init__.py [hunk 1]
```diff
@@ -155,8 +150,9 @@ def _detect_render_type(data, input_type) -> RenderType:
         return RenderType.IMAGE
     elif isinstance(data, itkwasm.PointSet):
         return RenderType.POINT_SET
-    elif isinstance(data, MultiscaleSpatialImage):
-        return RenderType.IMAGE
+    elif HAVE_MULTISCALE_SPATIAL_IMAGE:
+        if isinstance(data, MultiscaleSpatialImage):
```
[the remark this answers is not in the document]
Comment (thewtex) [reply itkwidgets/integrations/__init__.py:157]: :+1:

File: itkwidgets/integrations/__init__.py [hunk 2]
```diff
@@ -138,10 +138,12 @@ def _get_viewer_point_sets(point_sets):
         import torch
         if isinstance(point_sets, torch.Tensor):
             return point_sets.numpy()
-    if isinstance(point_sets, xr.DataArray):
-        return xarray_data_array_to_numpy(point_sets)
-    if isinstance(point_sets, xr.Dataset):
-        return xarray_data_set_to_numpy(point_sets)
+    if HAVE_XARRAY:
+        import xarray as xr:
```
Comment (bnmajor) [inline itkwidgets/integrations/__init__.py:142]: Just need to remove this colon and it looks good!

**Review by bnmajor — CHANGES_REQUESTED** [review]

File: itkwidgets/integrations/__init__.py [hunk 3]
```diff
@@ -155,8 +152,10 @@ def _detect_render_type(data, input_type) -> RenderType:
         return RenderType.IMAGE
     elif isinstance(data, itkwasm.PointSet):
         return RenderType.POINT_SET
-    elif isinstance(data, MultiscaleSpatialImage):
-        return RenderType.IMAGE
+    elif HAVE_MULTISCALE_SPATIAL_IMAGE:
+        from multiscale_spatial_image import MultiscaleSpatialImage:
```
Comment (bnmajor) [inline itkwidgets/integrations/__init__.py:156]: Need to remove the colon from the import

**Review by bnmajor — CHANGES_REQUESTED** [review]

**Comment by bnmajor** [comment]
@thewtex We also need an another import for xarray at [line 196](https://github.com/InsightSoftwareConsortium/itkwidgets/blob/c50aa00d7e1cd4e265935a31bab7241e15ea0a7b/itkwidgets/integrations/__init__.py#L196)

File: itkwidgets/integrations/__init__.py (hunk 3 above)
Comment (thewtex) [reply itkwidgets/integrations/__init__.py]: Fixed :-)

**Comment by thewtex** [comment]
> @thewtex We also need an another import for xarray at line 196

@bnmajor thanks, done!

**Comment by bnmajor** [comment]
@thewtex Looks great, thanks so much for this fix!! :fireworks:

**Comment by thewtex** [comment]
@bnmajor thank you for your patient testing! Let's march forward, and continue to march forward with improved CI :heavy_check_mark: :walking:

**Pull request merged**